database.go

```diff
@@ -332,7 +369,7 @@ func (m *Measurement) idsForExpr(n *influxql.BinaryExpr) (seriesIDs, bool, influ
 	}
 
 	// ignore time literals
-	if _, ok := value.(*influxql.TimeLiteral); ok {
+	if _, ok := value.(*influxql.TimeLiteral); ok || name.Val == "time" {
```

Comment:
How could this happen? Either it's a `TimeLiteral` or not, right? Also doing a string-compare? Is that necessary?